nx-X11/programs/Xserver/os/connection.c

```diff
@@ -988,7 +988,7 @@ CloseDownConnection(ClientPtr client)
     OsCommPtr oc = (OsCommPtr)client->osPrivate;
 
     if (FlushCallback)
-	CallCallbacks(&FlushCallback, NULL);
+	CallCallbacks(&FlushCallback, client);
```

Comment:
Oh dear. Sorry for this flaw. You are, of course, right on this. Amended locally already.